llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -213,26 +213,38 @@ getScalarSpecConstMetadata(const Instruction *I) {
 /// Recursively iterates over a composite type in order to collect information
 /// about its scalar elements.
 void collectCompositeElementsInfoRecursive(
```

Comment:
We need to have a unit-test for this change in the pass. Technically I can do that by looking into a properties file generated by `sycl-post-link`, but non-scalar properties are encoded using base64, so, it won't be an expressive test. However, this will be my plan if I didn't think of anything better. Any ideas are welcome

Reply:
I assumed the fix is needed urgently and the test will follow shortly, so I approved.

My idea would be adding logging to the spec constants pass and verifying the log output is expected. IIRC, some of passes use this approach.